wiki/Tutorials/ JupyterHub/JupyterHub.md

```diff
@@ -0,0 +1,190 @@
+helm list -A
+```
+
+* We can now add JupyterHub helm charts:
+
+```console
+helm repo add jupyterhub https://jupyterhub.github.io/helm-chart/
+helm repo update
+```
+
+### shares
+
+The JupyerHub in the example will include an extra storage that can be accessed by all users. This can be usefull to share files with all user. Here we use NFS, since this can be easily installed via OpenStack. To create an NFS share in the de.NBI node see [here](https://cloud.denbi.de/wiki/Compute_Center/Berlin/#create-a-nfs-share). If you use a different cloud setup, other storage options can also be integrated. See here (k8s website) to define specific shares. In the example below a 1000GB NFS share is created. To use NFS in the de.NBI cloud the two fields `server` and `path` have to be filled in. The content below needs to be stored as `data-pv.yaml` on the deploy-node.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: data
+spec:
+  capacity:
+    storage: 1000Gi
+  accessModes:
+    - ReadWriteMany
+  nfs:
+    server: #TODO: set url to NFS server (e.g. manila-dmz-prod.isi.denbi.bihealth.org)
+    path: #TODO: set path to NFS share
+  mountOptions:
+    - nfsvers=3
+```
+
+* The next command creates the share in Kubernetes:
+
+```bash
+kubectl apply --namespace jhub -f data-pv.yaml
+```
+
+* To mount the share into JupyterHub, we also need a persistent volume claim. The content below needs to be stored in the file `data-pvc.yaml`
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: data-pvc
+spec:
+  accessModes:
+    - ReadWriteMany
+  storageClassName: ""
+  resources:
+    requests:
+      storage: 1000Gi
+```
+
+* Finally, the next command creates the volume claim in Kubernetes:
+
+```bash
+kubectl apply --namespace jhub -f data-pvc.yaml
+```
+
+### Network
+
+In the Berlin node, a floating-ip with public access needs specifically be assigned to your project. Please ask the Berlin cloud admin team for the provision of a public floating ip via mail: denbi-cloud@bih-charite.de . We will use the floating IP `xxx.xxx.xxx.xxx` in the example and the respective public IP is `xxx.xxx.xxx.xxx`. Both adresse are just used for the example and should not really exist.
+
+### DNS & encryption
+
+To access the portal from the internet via https, we also need a DNS entry and a TLS certificate. In some cases, a DNS entry for the bihealth.org domain can be requested for a de.NBI project at the Berlin site. Please contact the site cloud admins to get information for your specific project. In the example here, we use the URL `my_platform.bihealth.org`. Next, we also need a certificate for encrypted communication. For example, a certficate can be created by letsencrypt (https://letsencrypt.org/de/getting-started/). The keyfile and the certificate file need to be stored on the `deploy-node`.  Both files are stored as a secret in Kubernetes so that we can use them in the JupyterHub deployment. Name of the secret is `jupyter-tls`.
```

Comment:
```suggestion
To access the portal from the internet via https, we also need a DNS entry and a TLS certificate. In some cases, a DNS entry for the bihealth.org domain can be requested for a de.NBI project at the Berlin site. Please contact the site cloud admins to get information for your specific project. In the example here, we use the URL `my_platform.bihealth.org`. Next, we also need a certificate for encrypted communication. For example, a certificate can be created by [Let's Encrypt](https://letsencrypt.org/de/getting-started/). The keyfile and the certificate file need to be stored on the `deploy-node`.  Both files are stored as a secret in Kubernetes so that we can use them in the JupyterHub deployment. Name of the secret is `jupyter-tls`.
```